libs/core/langchain_core/utils/mustache.py

```diff
@@ -425,11 +429,13 @@ def _get_partial(name: str, partials_dict: Dict[str, str]) -> str:
 #
 g_token_cache: Dict[str, List[Tuple[str, str]]] = {}
 
+EMPTY_DICT: MappingProxyType[str, str] = MappingProxyType({})
+
 
 def render(
     template: Union[str, List[Tuple[str, str]]] = "",
-    data: Dict[str, Any] = {},
-    partials_dict: Dict[str, str] = {},
+    data: Mapping[str, Any] = EMPTY_DICT,
```

Comment:
should we just make this Optional and default to None?

Reply:
That's the other solution which was used before MappingProxyType was introduced.
I think that it's not as good to use Optional as from a type perspective, it allows to pass None for no good reason.
But if you prefer Optional, I don't mind changing this.